walkingastro.lic

```diff
@@ -0,0 +1,210 @@
+custom_require.call %w[events]
+
+class WalkingAstro
+  def initialize
+    settings = get_settings
+    @astrology_prediction_skills_survival = settings.astrology_prediction_skills['survival']
+    main_loop
+  end
+
+  def get_telescope
+    echo "Getting telescope"
+    fput("stop play")
+    case DRC.bput("get my telescope","You get a", "You need a free hand to pick that up.", "You are already holding that.", "You should stop practicing your Athletics skill before you do that")
+    when "You get a"
+      determine_time
+    when "You need a free hand to pick that up."
+      quick_retry
+    when "You are already holding that."
+      determine_time
+    when "You should stop practicing your Athletics skill before you do that"
+      return
+    end
+  end
+
+  def quick_retry
+    echo "Trying again quickly!"
+    pause 20
+    get_telescope
+  end
+
+  def determine_time
+    echo "Determining time"
+    case DRC.bput("time","it is dawn", "morning", "midday", "noon", "afternoon", "dusk", "sunset", "evening", "midnight", "night", "almost dawn", "approaching sunrise")
+    when "it is dawn"
+      day_center
+    when "morning"
+      day_center
+    when "midday"
+      day_center
+    when "noon"
+      day_center
+    when "afternoon"
+      day_center
+    when "dusk"
+      day_center
+    when "sunset"
+      night_center
+    when "evening"
+      night_center
+    when "midnight"
+      night_center 
+    when "night"
+      night_center
+    when "almost dawn"
+      night_center
+    when "approaching sunrise"
+      night_center
+    end
+  end
+
+  def day_center
+    echo "Day center"
+    fput("center telescope on sun")
+    while line = get
+      if line.include?("That's a bit tough to do when you can't see the sky")
+        echo "Inside, next cycle"
+        stow_telescope
+        return
+      end
+      if line.include?("Your search for the sun")
+        echo "Switch to Heart"
+        night_center
+      end
+      if line.include?("You are a bit too busy performing to do that.")
+        echo "Playing an instrument, retrying"
+        fput("stop play")
+        day_center
+      end
+      observe
+      return
+    end
+  end
+
+  def night_center
+    echo "Night center"
+    fput("center telescope on heart")
```

Comment:
❔ Why 'heart' in specific?

Reply:
Heart is the only Survival night time constellation available all year, with no rising/setting during the night at unspecified times. It, like the sun during the day, is always available at night. As an added bonus, it fills the same pool that the sun does, which allows for the pool to fill in a timely manner for predictions.

Reply:
👍 thank you for the explanation